aten/src/ATen/native/sparse/cuda/SoftMax.cu

```diff
@@ -13,6 +13,21 @@
 #include <ATen/native/sparse/cuda/SparseCUDAApplyUtils.cuh>
 #include <ATen/native/sparse/cuda/SparseCUDABlas.h>
 
+#ifndef AT_PER_OPERATOR_HEADERS
+#include <ATen/Functions.h>
+#include <ATen/CUDAFunctions.h>
+#include <ATen/NativeFunctions.h>
+#else
+#include <ATen/ops/_masked_softmax_native.h>
+#include <ATen/ops/_log_softmax_cuda_dispatch.h>
+#include <ATen/ops/_log_softmax_backward_data_cuda_dispatch.h>
+#include <ATen/ops/_softmax_cuda_dispatch.h>
+#include <ATen/ops/_softmax_backward_data_cuda_dispatch.h>
+#include <ATen/ops/equal_native.h>
+#include <ATen/ops/full.h>
+#include <ATen/ops/softmax.h>
```

Comment:
I used iwyu for all the `.cpp` files which is why that PR is in this stack. Unfortunately iwyu doesn't work with CUDA so the `.cu` files have to be manually looked through to find the required headers.